datastore-design/go-grpc-demo/protobuf/datastore.proto

```diff
@@ -10,7 +12,7 @@ option go_package = "./datastore";
 //   time.
 // - An 'observation' belongs to a time series and contains obs time, obs value,
 //   and metadata that depend on obs time.
-// - All obs times are Unix timestamp (i.e. relative to Unix epoch)
+// - All obs times are of type google.protobuf.Timestamp
```

Comment:
What's the rationale of using Timestamp instead of int64?
Unless we are required to support sub-second obs times (not sure about this I must admit!) it seems like int64 would be a simpler and more efficient representation, but maybe only just very slightly, so not really a big deal. I may have overlooked something, though.

Reply:
The rationale is to support sub-second obs times, and to make it more explicit in the protobuf schemas that this is a time stamp (and thus potentially make conversion to datetime in any client easier).

I was surprised to see only second accuracy, and looked into adding sub-second accuracy, and found the usual way was to use Timestamp.

If we don't want sub-second accuracy, there is not much point to this PR.

Reply:
I think we should be general already from the start. See other comment. So the PR as such is fine I think!

Reply:
Also good point about ease of conversion etc. for type Timestamp. And also: I guess the storage and processing overhead of Timestamp is negligible compared to int64 anyway.